src/main/java/com/softserve/teamproject/service/impl/GroupServiceImpl.java

```diff
@@ -105,4 +88,28 @@ public void deleteGroup(int groupId, String userName) {
 
     groupRep.delete(group);
   }
+
+  /**
+   * Returns groups with filter.
+   *
+   * @param wrapper contains filter values
+   * @return groups with filter
+   */
+  public Iterable getGroupsByFilter(GroupsFilterRequestWrapper wrapper) {
+    List<BooleanExpression> predicates = new ArrayList<>();
+
+    if (wrapper.getLocations() != null) {
+      predicates.add(GroupExpressions.getByLocationIds(wrapper.getLocations()));
+    }
+
+    if (predicates.size() > 0) {
+      BooleanExpression predicate = predicates.get(0);
+
+      for (int i = 1; i < predicates.size(); i++) {
+        predicate = predicate.or(predicates.get(i));
```

Comment:
That code shouldn't be reachable. You are putting a single item in list on line 102. I can't see where you could get more that one entry in the list. So I feel like this code should be optimized. Please recheck this.